cohort-evaluator-spark/src/main/java/com/ibm/cohort/cql/spark/aggregation/ColumnRuleCreator.java

```diff
@@ -0,0 +1,108 @@
+package com.ibm.cohort.cql.spark.aggregation;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.ibm.cohort.cql.evaluation.CqlEvaluationRequest;
+import com.ibm.cohort.cql.library.CqlLibraryDescriptor;
+import com.ibm.cohort.cql.library.CqlLibraryProvider;
+import com.ibm.cohort.cql.spark.optimizer.DataTypeRequirementsProcessor;
+import com.ibm.cohort.cql.translation.CqlToElmTranslator;
+import com.ibm.cohort.cql.util.EqualsStringMatcher;
+import com.ibm.cohort.cql.util.StringMatcher;
+
+public class ColumnRuleCreator {
+	private static final Logger LOG = LoggerFactory.getLogger(ColumnRuleCreator.class);
+
+	private final List<CqlEvaluationRequest> requests;
+	private final CqlToElmTranslator cqlTranslator;
+	private final CqlLibraryProvider libraryProvider;
+	
+	public ColumnRuleCreator(List<CqlEvaluationRequest> requests, CqlToElmTranslator cqlTranslator, CqlLibraryProvider libraryProvider) {
+		this.requests = requests;
+		this.cqlTranslator = cqlTranslator;
+		this.libraryProvider = libraryProvider;
+	}
+	
+	/**
+	 *
+	 * @param context ContextDefinition whose CQL jobs will be interrogated for data requirements.
+	 * @param disableColumnFiltering If true, return null. Otherwise return a populated map
+	 * @return Map of data type to the fields in that datatype that are used by the CQL jobs.
+	 *         null is returned when column filtering is disabled.
+	 */
+	public Map<String, Set<StringMatcher>> getColumnRulesForContexts(ContextDefinition context, boolean disableColumnFiltering) {
+		Map<String, Set<StringMatcher>> pathsByDataType = null;
+		if( ! disableColumnFiltering ) {
```

Comment:
Do you have any thoughts on  a possible `ContextRetrieverFactory` class or something similar that could hold all of the logic needed to set up the context retriever to use filtering or not? I like the idea of the `SparkCqlEvaluator` only needing to know whether it wants filtering or not, and then "something else" handles building the correct `ContextRetriever` to use.

I agree the current form of the code has a structure somewhere in the middle  that isn't all that great.

Reply:
I think having a `ContextRetrieverFactory` is reasonable, but just moving the check to `SparkCqlEvaluator` seems like a reasonable compromise if it's just the one check you wanted to do there.

Reply:
I think I'm in line with Dan's thoughts.  Since the current boolean value is a single program argument that doesn't change at all, I don't see a need for a full on factory.  I'm down for one whenever we add even more logic around context retriever creation.

Reply:
Makes sense. I suppose having `SparkCqlEvaluator` know to pass `null` vs. an object isn't too clunky of an interface to have.